### ISSUE TITLE
Process Types and Transfer Types: good and bad in circular economy

### ISSUE THREAD

**Issue by fosterlynn** [valueflows/valueflows#96]
Interesting list from @pmackay around what they have developed as a list of process types and transfer types or exchange types.  And a way to include the concept of good > bad in terms of effect in the circular economy.

From Gitter:
pmackay
12:10
FYI, here's the list of exchange types and processes we've worked on for the circular economy (no doubt there are more):
Renting
Borrowing
Lending
Reuse
Repairing
Donating
Selling
Giving / Gifting
Swapping
Repurposing
Refurbish
Upcycling
Recycling
Closed loop recycling
Downcycling
Composting
Home composting
Home digestion
Anerobic digestion
Disposal
Landfill
Pyrolisis
Gasification
fosterlynn
12:14
@pmackay cool, studying.

    Do you think its possible to have a common set of Types defined? 

Sometimes it is and sometimes is isn't, because it is important for emerging groups to work through how they want to define the next economy, there are lots of experiments in this transitional stage.
(We had a lot of discussion on this in Agent definition, because @elf-pavlik coming out of LOD perspective wanted to try to define the Agent Relationship Types. Which I get. But we just aren't ready to do that.)
@pmackay looks like your list is partially specific to your domain, and partially pretty general. Still studying....
bhaugen
12:28
@pmackay nice list, though, I like it!
pmackay
12:29
basically all methods of what can done with a thing, going from good to worse in terms of levels of circular economy. Reduce/reuse at top, dispose at bottom, etc.
bhaugen
12:29
@/all - first draft of https://github.com/valueflows/valueflows/wiki/Provenance
more to come, later today maybe
fosterlynn
12:30
thanks @bhaugen ; @pmackay i didn't mean to be critical! i like your list too!
bhaugen
12:30
good-to-worse is an interesting relationship among process types
I wonder if that will have analogs in other domains?
pmackay
12:31
of course its only a good > bad relationship in context of circular economy and waste hierarchy

**Comment by pmackay** [comment]
@fosterlynn @bhaugen so how could we start to develop the list above into more of a set of RDF terms?

**Comment by fosterlynn** [comment]
@pmackay I'll take some time this afternoon to work on that - at least for a discussion starter.  I am still not fluent in RDF, but certainly better at it than a few months ago.  (Credit to @elf-pavlik for working hard at educating us newbies!)  

So part of this task is making RDF and part of it is figuring out the model.  Do you have anything you could point us to as to how you use the list?  Or how that data relates to your other data?  If you have a data or class model, or some code where I can derive a model, I'd be interested.

**Comment by pmackay** [comment]
I'd be happy to build the RDF. The question I'm not sure about is what approach to use to define a list. Any thoughts on the two methods I suggested in gitter?

> So we might try to define a SKOS of exchange types. If we wanted to capture buy, sell, donate, accept as 4 exchange types, we could define them all as VF terms (where vf = valueflows, gr = Good Relations):
> vf:buy
> vf:sell
> vf:donate
> vf:accept
> 
> Then use RDF to say
> vf:buy === gr:Buy
> vf:sell === gr:Sell
> 
> Or just define
> vf:donate
> vf:accept
> and simply use
> gr:Buy
> gr:Sell

**Comment by fosterlynn** [comment]
How about something like this, brainstorming more than proposing:

vf:Exchange
vf:Sale rdfs:subClassOf vf:Exchange
vf:Donation rdfs:subClassOf vf:Exchange
vf:Rental rdfs:subClassOf vf:Exchange
etc.

vf:Process
vf:Refurbish rdfs:subClassOf vf:Process
vf:Compost rdfs:subClassOf vf:Process
etc.

(Ignoring GR for a moment, would have to look them up again, but totally want to use existing vocabs where we can.)

Note that buy and sell is the same thing, just from different perspectives.  In the model I'm trying to work out right now in code, that perspective split happens down at the flow, where we have a give flow and a receive flow for each transfer.  There might be 2 transfers in an exchange, like in a sale there is a transfer of goods, and a reciprocal transfer of money, something like Delivery and Payment.  The delivery would have a give (you give me the goods) and a receive (I receive the goods).  The payment would be me giving you money and you receiving money from me.  Even though that is also one thing, this is needed to create the value flows. Like:

Resource > input > Process > output > Resource > give > Transfer > receive > Resource

Issue: Donation for example has only one transfer, and there really isn't an exchange, so that might mess with the above subclassing.  In NRP so far, I'm keeping exchanges even where there is only one transfer, for historical reasons in the code and it is cleaner to handle in general.  But it doesn't seem really right.

**Comment by pmackay** [comment]
This sounds good. You're modelling the Exchanges as nouns here, e.g. I have a sale. However in GR they are verbs, e.g. I sell this thing. So is that a fundamental difference? That in VF you need/want to model the exchange as a thing, rather than a "function" as GR might define it?

**Comment by fosterlynn** [comment]
Yes, although I waffled around nouns vs verbs actually.  Where I got to (for discussion) is that Exchange (and Transfer) and Process are nouns.  They contain a bunch of stuff, and the verbs happen inside.  I'm trying to get my head into nouns and verbs as nodes and edges in LOD land.  So a Process or a Transfer is a noun (node), and the ValueFlow (EconomicEvent) is a verb (edge).  And the Resource is a noun, I guess duh.  The flows take resources in and out of transfers or processes.

Here is the GR definition of BusinessFunction:

> The business function specifies the type of activity or access (i.e., the bundle of rights) offered by the gr:BusinessEntity on the gr:ProductOrService through the gr:Offering. Typical are sell, rental or lease, maintenance or repair, manufacture / produce, recycle / dispose, engineering / construction, or installation.

They do seem more verb-like, but perhaps not clearly settled on verbs?  I think a bundle of rights is a noun, but selling a bundle of rights is a verb, so hmmm.  Rental is a noun, rent would be a verb.

As an aside, GR is focused on the perspective of an enterprise, so they don't worry about buy vs sell type of thing.  We are focused at the perspective of a network.

**Comment by pmackay** [comment]
Agree the GR stuff is a bit mixed, but mostly verbs I think. And yes there is a difference. Also how to approach the difference between expressing an offer (as GR or schema.org does it) compared with capturing the value flow in VF? Intent vs record of exchange/transfer? A bike shop has an offer/intent to “sell” bikes. An exchange happens and is recorded when a “sale” is made. Are both needed?

**Comment by fosterlynn** [comment]
VF should have both.  There's a whole corresponding layer of plan/offer/commit/etc. "above" the reality layer.  And when a sale happens, it should record its connection to the offer. 

[more later....]

**Comment by fosterlynn** [comment]
Thinking further, there is even more separation.  But I still think VF should have all of this, its all part of conducting economic activity in a networked way.

So, an offer can be made broadly or not.  An offer could be just: I have XYZ resource type to sell (and I either have resources of that type or can get or make them if you order some).  Down to: I have a bike with a dodgy brake to give away to someone in my local network.

But beyond any kind of offer, there can be a more specific commitment to carry out the exchange that could be recorded before the exchange happens.  That would involve both agents, not just the offering one.  Like I want to order 5 of those X's; or I'd like to have your bike and I'll meet you tomorrow at 5pm at your place.  NRP supports commitments but not really the intent side.

VF has a separate Intent repo where we were hoping to have some conversations with people who are doing intents (both offers and wants), like the time bank and mutual credit people.  Or really anyone who has a way to offer stuff or services out on the web.  Haven't tried to get anything going yet but we should when any of us (including you of course) has a need.  

I'll try for a picture, although it will probably start off as UML.  :)

**Comment by bhaugen** [comment]
On the [intent](https://github.com/valueflows/intent) readme, there's a picture of a funnel loop depicting repeated exchanges between the same agents.

That's also the picture of repeated conversations for action, which is what is kicked off then offer meets want, which has a picture on the [exchange](https://github.com/valueflows/exchange) readme.

What we've seen in mutual credit communities, where offers and wants are used a lot, is that the resource types for wants and sometimes offers are expressed in terms of a fairly broad level of a taxonomy.

We think they will evolve into faceted categorization, but have not done so yet. If they do, probably each facet will have its own taxonomy.

**Comment by fosterlynn** [comment]
@pmackay @bhaugen I veered the intent train of thought into a new issue #98 .

**Comment by fosterlynn** [comment]
Here is another attempt to see how the circular economy list could fit into the exchange - transfer - flow paradigm.  Just trying out types of exchanges and types of transfers out for size, names can be adjusted to taste.  And some types of processes while we're here.

So Exchange has Exchange - Transfer - ValueFlow, Process has just Process - ValueFlow. 

Renting (subclass of Exchange)
-- Possession (subclass of Transfer)
---- Give (subclass of ValueFlow)
---- Receive (subclass of ValueFlow)
-- Payment (subclass of Transfer)
---- Give (subclass of ValueFlow)
---- Receive (subclass of ValueFlow)
Lending (subclass of Exchange)
-- Lend/Borrow (subclass of Transfer)
---- Give (subclass of ValueFlow)
---- Receive (subclass of ValueFlow)
-- Return (subclass of Transfer)
---- Give (subclass of ValueFlow)
---- Receive (subclass of ValueFlow)
Selling (subclass of Exchange)
-- Delivery or Shipment (subclass of Transfer)
---- Give (subclass of ValueFlow)
---- Receive (subclass of ValueFlow)
-- Payment (subclass of Transfer)
---- Give (subclass of ValueFlow)
---- Receive (subclass of ValueFlow)
Giving / Gifting (subclass of Exchange)
-- Gifting (subclass of Transfer)
---- Give (subclass of ValueFlow)
---- Receive (subclass of ValueFlow)
Swapping (subclass of Exchange)
-- Transfer (or maybe there's a good subclass)
---- Give (subclass of ValueFlow)
---- Receive (subclass of ValueFlow)
-- Transfer(or maybe there's a good subclass)
---- Give (subclass of ValueFlow)
---- Receive (subclass of ValueFlow)
Donating (subclass of Exchange)
-- Donating (subclass of Transfer)
---- Give (subclass of ValueFlow)
---- Receive (subclass of ValueFlow)

Repairing (subclass of Process)
---- ToBeChanged (subclass of Input, which is subclass of ValueFlow) like the dodgy bike
---- Consume and/or Use (subclass of Input, which is subclass of ValueFlow) like parts or equipment
---- Change (subclass of Output, which is ValueFlow) the bike repaired
Refurbish (subclass of Process)
---- probably like Repairing?
Closed loop recycling (subclass of Process)
---- ?
Composting (subclass of Process)
---- Consume (subclass of Input, which is subclass of ValueFlow)
---- Use (subclass of Input, which is subclass of ValueFlow) like maybe a compost bin
---- Produce (subclass of Output, which is subclass of ValueFlow)
Home composting (subclass of Process) same as Composting?
Home digestion (subclass of Process) ?
Anerobic digestion (subclass of Process) 
Landfill (subclass of Process)
Pyrolisis (subclass of Process)
Gasification (subclass of Process)

Recycling (unclear if Process or Exchange)
Reuse (unclear if Process or Exchange)
Upcycling (unclear if Process or Exchange)
Downcycling (unclear if Process or Exchange)
Repurposing (unclear if Process or Exchange)
Disposal (unclear if Process or Exchange)

**Comment by samuelandert** [comment]
Hi Lynn, 
here my opinions on your questions:
Re- Up- and Downcycling are processes, since it will need an actor to manipulate objects from one state to the other. 
Reuse and repurposing are processes as well.
Disposal though is an exchange, since an object will change the actor. f.e. from person to organization or from person to object or location, such as a bin or wastedump.

**Comment by samuelandert** [comment]
Home composting is same as composting - I don't see the need to differentiate here
Home digestions is same as digestion - again I don't see the need to differentiate here
refurbish is same as repairing

**Comment by fosterlynn** [comment]
@Samsemilio hey, thanks and hello!  You're comments make sense.  I'll let @pmackay respond in detail, since the list came from him..... or do you work with Paul and @djwesto ?

Yeah, the agents are interesting on things like Disposal.  How to account for externalized costs?  Which we do want to do!  Maybe the agent on the receive side of Disposal is the community, or the natural ecosystem or something, depending on where the resource was disposed of?

**Comment by samuelandert** [comment]
I am coming from elf, he was talking about value flows, so I had a look ;)
I am defining disposals as something which is not recyclable and needs a physical place to be stored for the long term until technologies are able to recycle or a new actor is picking up the previously disposed materials for repurposing. So the final receiving end of an disposal would be a location. 
But you can imagine a flow like:
disposal -> transfer to -> garbage collector/organisation/community/ ->  transfer -> location

**Comment by fosterlynn** [comment]
> I am coming from elf, he was talking about value flows, so I had a look

Awesome, and thanks for chiming in!

**Comment by pmackay** [comment]
@Samsemilio thanks for comments :)

> Home composting is same as composting

Difference I see is the process. Home composting, people keep their organic waste locally. General composting typically managed by council means organics are moved to central facility. In North London the large composting facility then will offer tonnes of compost to community groups. Very different value flows :)

> refurbish is same as repairing 

Here I see refurbish as potentially making something new/better that is different to original, e.g. putting a new cover on an old sofa. Repair is restoring an item to its original state, e.g sewing a rip in the same sofa cover. But more opinions are useful and is it important to distinguish between those?

@fosterlynn about your list above, the top level items, e.g Renting are written as verbs but then the sub items representing value flows are also verbs. Would the top level work better as nouns? I'm also unsure if the VF sub elements are totally separate records or just properties, i.e. a record of a Rental simply assumes the give/receive required for someone to rent and return something.

**Comment by samuelandert** [comment]
@pmackay: I see, differentiating between refurbish and repairing makes sense :) 
How would you then differentiate between upcycle an refurbish? 

The process of composting itself always is the same. The only difference would be the location, the actors and the scale. Therefore, I am defining the movement of organics to a central facility more as an exchange than a process.

**Comment by pmackay** [comment]
@Samsemilio upcycle is harder, I see that as a more radical change, e.g. turning advertising banners into new bags. But other examples might blur the lines.

Agree composting process the same. This is where I'm struggling a bit to see the right way to define the distinctions between process, exchange, etc.

**Comment by bhaugen** [comment]
Might want some different process types for different types of repair, refurbish, upcycle, etc.

For example, repair of some objects may be involve diagnosis, disassembly, getting new parts, and re-assembly.

Re-covering a sofa or repairing shoes or clothing would be pretty different.

Those differences will emerge in practice in real situations, though. Probably too speculative to define them upfront.

**Comment by fosterlynn** [comment]
> the top level items, e.g Renting are written as verbs but then the sub items representing value flows are also verbs. Would the top level work better as nouns?

Yes, sorry about that, I just copy pasted your list and then was thinking only about the structure.... let's fix it.

> I'm also unsure if the VF sub elements are totally separate records or just properties, i.e. a record of a Rental simply assumes the give/receive required for someone to rent and return something.

I'll try making it more complete with all the properties and then we can see how it pans out.  In our software those are different records (objects that persist in relational tables): Exchange, Transfer, ValueFlow (which we call EconomicEvent).  I'll give a whirl at an LOD representation of some sort.  (Unsure of timing, but will do it.)

**Comment by bhaugen** [comment]
> the right way to define the distinctions between process, exchange, etc.

An exchange or transfer involves at least two agents where the same resource is given by one agent and taken by another.

A process converts input resources to different output resources. Even if it's a repair, where the same bike is an input and an output, the bike is different (repaired) as an output.

An externality like waste water is usually also different from the water that went into the process: e.g. polluted.

**Comment by fosterlynn** [comment]
@pmackay @Samsemilio @bhaugen great discussion on the distinctions.  To me this highlights the need to always allow for user definable capability on types of processes and types of exchanges.  Even leaving out the capability to use process types to define specific recipes for creating things, there is an ongoing tension / contradiction between nailing down the type or subclass vocab vs leaving it to users to define:

The more that different groups can agree on for some of these basic or high level types, the better they can communicate.  And the better people can define things like better or worse types of processes in terms of the circular economy, as an example.

But even for the high level types, we are in a time of economic experimentation, so I think we need to have enough flexibility so people do not feel constrained as they develop their own sets of processes and exchanges.

I picture it as an ongoing process where as people and groups get more practice with their own definitions, and then can converge through discussion or proposals of sets of vocab, and terms will come to have more standard definitions over time.  

(The original valueflows gang had a whole bunch of discussion on this topic around types of agent relationships, and also types of agents.  We decided to define a layer of vf subclasses where there is agreement, and let groups define their own subclasses of those, or use other vocabs if the it fits for them.  Never nailed down the vf ones though.  :smile: )

**Comment by pmackay** [comment]
@fosterlynn actually you've reminded me of the diagrams at the bottom of https://github.com/valueflows/valueflows. So should the list I posted be framed more as exchange/process types, i.e. a separate hierarchy/taxonomy, rather than being subclasses of Exchange/Process?

Does that also depend on what properties need associating with different sub-types of Exchange/Process?

**Comment by pmackay** [comment]
Also just noticed that the [TradeAction](https://schema.org/TradeAction) and [TransferAction](https://schema.org/TransferAction) types on schema.org cover a lot of the common exchange types - what are your thoughts on how useful and usable those are?

**Comment by bhaugen** [comment]
We've had a lot of discussion about which existing vocabulary elements to use. In general, everybody wants to do that. But the consensus seems to be to do it _after_ we work through a lot of use cases and know better what we need.

One of the problems with schema.org (which seems like a hodgepodge to me) and goodrelations (which I like a lot) is that their goals are so different. I think schema.org originated for search optimization, and GR for e-commerce. E-commerce is part of our purview here, but is a small part of economic networks. Especially economic networks aimed at social change.

**Comment by fosterlynn** [comment]
> should the list I posted be framed more as exchange/process types, i.e. a separate hierarchy/taxonomy, rather than being subclasses of Exchange/Process?

@pmackay welcome to the depths of valueflows!  :)

The diagrams come from NRP, where we make heavy use of type objects rather than subclassing partly because of object-relational issues, but mostly because we need them to be user definable, which can't be done by subclassing in object-land.  As we move the model to LOD, so far we are moving out of the type pattern and more towards subclasses and subproperties.  But with something like types of processes, we will also run into problems with subclassing because of the quantity (I think).  Same might happen with resource types, which can be general or very specific, as in a product definition. 

Your thoughts will be very useful here because you have LOD experience!  Along with @elf-pavlik of course.... :)

**Comment by fosterlynn** [comment]
> Also just noticed that the TradeAction and TransferAction types on schema.org

I basically agree with @bhaugen on these, and on the process we have been following.  At the same time, I find it useful to browse the existing vocabs anyway in case they provide vocab or just an understanding of how people have been thinking of the concepts.  There also is an existing vocab wiki page where we can keep references for later: https://github.com/valueflows/valueflows/wiki/Useful-vocabularies.  Pretty general now, but could be also made more specific.
